### PR TITLE
update conversion script to support permissions

### PR DESCRIPTION
**Pull request by mattcasey**

### CODE REVIEW

File: scripts/migrations/2023_11_30_convertProposals.ts [hunk 1]
```diff
@@ -38,21 +47,38 @@ async function convertProposals() {
               }
             });
           }
+          await tx.proposalEvaluation.create({
```
Comment (motechFR) [inline scripts/migrations/2023_11_30_convertProposals.ts:50]: General suggestion for preceding statement, isPublished would only be proposal.status !== 'draft'

File: scripts/migrations/2023_11_30_convertProposals.ts [hunk 2]
```diff
@@ -38,21 +47,38 @@ async function convertProposals() {
...
-                create: {
-                  proposalId: p.id
-                }
+                create: p.category?.proposalCategoryPermissions
+                  .filter((perm) => perm.permissionLevel === 'full_access')
+                  .map((perm) => ({
+                    proposalId: p.id,
+                    roleId: perm.roleId,
+                    systemRole: perm.spaceId ? 'space_member' : undefined
+                  }))
```
Comment (motechFR) [inline scripts/migrations/2023_11_30_convertProposals.ts:78]: Looks good!

File: scripts/migrations/2023_11_30_convertProposals.ts [hunk 3]
```diff
@@ -38,21 +47,38 @@ async function convertProposals() {
...
+              },
+              permissions: {
+                create: getDefaultPermissions()
```
Comment (motechFR) [inline scripts/migrations/2023_11_30_convertProposals.ts:81]: makes sense

File: scripts/migrations/2023_11_30_convertProposals.ts [hunk 4]
```diff
@@ -38,21 +47,38 @@ async function convertProposals() {
...
               result,
               completedAt: result ? completedAt || new Date() : null,
               proposalId: p.id,
               snapshotId: p.page?.snapshotProposalId,
               voteId: vote?.id,
               reviewers: {
-                create: {
-                  proposalId: p.id
-                }
+                create: p.category?.proposalCategoryPermissions
```
Comment (motechFR) [inline scripts/migrations/2023_11_30_convertProposals.ts:72]: Shouldn't this be createMany? Haven't used create before for multiple items

**Comment by motechFR** [comment]
Final potential issue: Are we missing setting currentEvaluationId on the proposal? Can't see it here

File: scripts/migrations/2023_11_30_convertProposals.ts (hunk 4 above)
Comment (mattcasey) [reply scripts/migrations/2023_11_30_convertProposals.ts:72]: i actually dont know i was surprised prisma lets me

File: scripts/migrations/2023_11_30_convertProposals.ts (hunk 1 above)
Comment (mattcasey) [reply scripts/migrations/2023_11_30_convertProposals.ts:50]: thanks! I updated isPublished to only check for draft. THis changed over time

**Pull request merged**